lib/inc/internal/facts/resolvers/operating_system_resolver.hpp

```diff
@@ -216,6 +216,20 @@ namespace facter { namespace facts { namespace resolvers {
          * @return Returns the resolver data.
          */
         virtual data collect_data(collection& facts);
+
+        /**
+         * Collects the resolver's kernel data.
+         * @param facts The fact collection that is resolving facts.
+         * @param result The current resolver data.
+         */
+        virtual void collect_kernel_data(collection& facts, data &result);
```

Comment:
They have to be protected for the inheriting class to override them. The usage here makes sense.